### PR TITLE
Accept string value in Span status setter

### PR DESCRIPTION
**Pull request by B-Step62**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/B-Step62/mlflow/pull/11585?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/11585/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 11585
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

Allow passing string value for `set_status` method for convenience.

```
with mlflow.start_span("test") as span:
    span.set_status("OK")
```

### How is this PR tested?

- [x] Existing unit/integration tests
- [x] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [x] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by BenWilson2** [comment]
Are we going to update the Client APIs `end_trace` and `end_span` to accept `str` in a follow-up PR, or were you going to add that to this PR?

**Comment by B-Step62** [comment]
> Are we going to update the Client APIs end_trace and end_span to accept str in a follow-up PR, or were you going to add that to this PR?

They just call `set_status` of the wrapper so already accept `str`, but I haven't updated the docstring. Will do in this PR.

File: mlflow/tracking/client.py [hunk 1]
```diff
@@ -534,7 +533,7 @@ def end_trace(
         request_id: str,
         outputs: Optional[Dict[str, Any]] = None,
         attributes: Optional[Dict[str, Any]] = None,
-        status: SpanStatus = SpanStatus(TraceStatus.OK),
+        status: SpanStatus = "OK",
```
Comment (harupy) [inline mlflow/tracking/client.py:536]: can we update the type hint?

**Review by harupy — APPROVED** [review]
LGTM!

File: mlflow/tracking/client.py [hunk 2]
```diff
@@ -550,7 +549,10 @@ def end_trace(
             attributes: A dictionary of attributes to set on the trace. If the trace already
                 has attributes, the new attributes will be merged with the existing ones.
                 If the same key already exists, the new value will overwrite the old one.
-            status: The status of the trace. The default status is OK.
+            status: The status of the trace. This can be a
+                :py:class:`SpanStatus <mlflow.entities.SpanStatus>` object or a string representing
+                of the status code defined in :py:class:`TraceStatus <mlflow.entities.TraceStatus>`
```
Comment (BenWilson2) [inline mlflow/tracking/client.py:554]:
```suggestion
                the status code defined in :py:class:`TraceStatus <mlflow.entities.TraceStatus>`
```

File: mlflow/tracking/client.py [hunk 3]
```diff
@@ -726,7 +728,10 @@ def end_span(
             attributes: A dictionary of attributes to set on the span. If the span already has
                 attributes, the new attributes will be merged with the existing ones. If the same
                 key already exists, the new value will overwrite the old one.
-            status: The status of the span. The default status is OK.
+            status: The status of the span. This can be a
+                :py:class:`SpanStatus <mlflow.entities.SpanStatus>` object or a string representing
+                of the status code defined in :py:class:`TraceStatus <mlflow.entities.TraceStatus>`
```
Comment (BenWilson2) [inline mlflow/tracking/client.py:733]:
```suggestion
                the status code defined in :py:class:`TraceStatus <mlflow.entities.TraceStatus>`
```

**Review by BenWilson2 — APPROVED** [review]
LGTM once nits and https://github.com/mlflow/mlflow/pull/11585/files#r1549927308 is addressed!

**Review by dbczumar — APPROVED** [review]
LGTM once existing comments are addressed. Thanks @B-Step62 !

**Pull request merged**